### PR TITLE
Add Choose NPM Publish ADR

### PR DESCRIPTION
**Pull request by johnnolan**

### CODE REVIEW

File: adr/choose_npm_publish.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+This brings the following benefits
+
+- Other teams can contribute to the same source code
+- Reduces potential duplication of similar functionality across products
+- Gives back to the open source community
+- Get feedback from the wider community in making us better developers
+- Modularised code for cleaner codebases and easier testing
+
+The downside of this approach means development cycles may take longer. This is due to the extra work to build and publish the package. Long term the benefits you gain outweigh this initial cost.
```
Comment (mattmachell) [inline adr/choose_npm_publish.md:42]: Maybe break this up in a list like the above benefits for consistency? Also has the consequence of more repos to maintain and the potential for loss of ownership of a component if it moves out of the product teams zone of control.

File: adr/choose_npm_publish.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# Choose NPM Publish
+
+## Status
+
+Accepted
+
+## Context
+
+As we split our code in microservices on the front-end, we need to have a way of sharing common code across products.
```
Comment (mattmachell) [inline adr/choose_npm_publish.md:14]: Microservices on the front-end or micro-front-ends? Maybe also specify front-end/JS code across products?

File: adr/choose_npm_publish.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+We have this issue currently with a few features we use over many products. This will give developers an option to share these features with others.
```
Comment (mattmachell) [inline adr/choose_npm_publish.md:16]: Example might help context?

File: adr/choose_npm_publish.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+## Decision
+
+Where possible, we should use NPM to publish any reusable code in our projects and consume them.
+
+We already use NPM repositories for all our products. Integration isn't an issue when consuming the packages.
```
Comment (mattmachell) [inline adr/choose_npm_publish.md:22]: Phrasing here might be better as "we already use NPM repos within our products for dependencies"? 

Also do we want to talk NPM vs Yarn and potential consequences down the line?

File: adr/choose_npm_publish.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+There will be a learning curve to setting up the repositories. This has been investigated already and can built upon.
+
+An example of this can be found here in this package.
+
+[https://www.npmjs.com/package/@ministryofjustice/opg-performance-analytics](https://www.npmjs.com/package/@ministryofjustice/opg-performance-analytics)
+
+Using the above as a template we can have full CI/CD with granular control over releases.
```
Comment (mattmachell) [inline adr/choose_npm_publish.md:30]: Do we want to create a template GitHub repo for people building for NPM as well?

File: adr/choose_npm_publish.md (hunk 4 above)
Comment (johnnolan) [reply adr/choose_npm_publish.md:22]: Am rewording this now but the NPM vs Yarn I feel is outside of the scope. However I think mentioning consuming via NPM or Yarn somewhere to clarify explicitly that it is about the repository use and not the tooling.

File: adr/choose_npm_publish.md (hunk 5 above)
Comment (johnnolan) [reply adr/choose_npm_publish.md:30]: I was thinking that as I was typing, will add it to my list to do this week and then change this to point there afterwards

File: adr/choose_npm_publish.md (hunk 1 above)
Comment (johnnolan) [reply adr/choose_npm_publish.md]: Have done this and added a few more to the list

File: adr/choose_npm_publish.md (hunk 2 above)
Comment (johnnolan) [reply adr/choose_npm_publish.md]: Change this sentence now to be more specific to our current usage with examples too.

File: adr/choose_npm_publish.md (hunk 3 above)
Comment (johnnolan) [reply adr/choose_npm_publish.md]: Added some concrete examples

File: adr/choose_npm_publish.md (hunk 4 above)
Comment (johnnolan) [reply adr/choose_npm_publish.md]: Cleared the wording up on this

**Pull request merged**